### PR TITLE
test: identity provider selection rule edge cases (AM-2171)

### PR DESCRIPTION
**Pull request by sagarchauhan-qa**
## :id: Reference related issue.
[AM-2171](https://gravitee.atlassian.net/browse/AM-2171)

## :pencil2: A description of the changes proposed in the pull request
Cover two gaps in identity provider selection rules. Routing a matching user to the right provider, and a user matching no rule at all, are already covered in `login-flow-inline.jest.spec.ts` and are not duplicated here.

- **A rule that cannot be evaluated.** `selectionRuleMatches` catches the failure, logs a warning and returns false, so the provider is skipped rather than the sign-in failing outright — a typo in a rule silently excludes a provider. Covered from both sides: with a second provider present the sign-in succeeds and returns *that* provider's profile, and when the broken rule is the only provider the sign-in is refused as `invalid_user` rather than surfacing an internal error.
- **A provider carrying no rule.** The existing fallback test asserts only that a code came back, and the fixture user exists in both providers, so it would pass with selection rules ignored entirely. The test here reads the profile back, and the ruled provider holds the higher priority, so a pass can only mean it was excluded rather than beaten to it.

No existing files changed. The weaker fallback test is left as it is, with this one alongside.

## :memo: Test scenarios
See jira.

## :computer: Add screenshots for UI
n/a

## :books: Any other comments that will help with documentation
n/a

## :man: :woman:@mentions of the person or team responsible for reviewing proposed changes
@gravitee-io/am

[AM-2171]: https://gravitee.atlassian.net/browse/AM-2171?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: gravitee-am-test/specs/gateway/login-flow/idp-selection-rules.jest.spec.ts [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+let appMalformedWithFallback: any;
+let appMalformedOnly: any;
+let appUnruledAlongsideRuled: any;
+
+/**
+ * Both inline providers hold the same username with a different profile, so signing in and
+ * reading `given_name` back says which provider actually answered. In every application below
+ * the ruled provider carries the higher priority, so it would answer if its rule were ignored.
+ */
+const signInAndReadGivenName = async (app: any): Promise<string> => {
```
Comment (sclarkamlab) [inline gravitee-am-test/specs/gateway/login-flow/idp-selection-rules.jest.spec.ts]: this is almost identical to signInAndReadProfile from https://github.com/gravitee-io/gravitee-access-management/pull/8456/changes can we reuse the existing function

Comment (sagarchauhan-qa) [reply gravitee-am-test/specs/gateway/login-flow/idp-selection-rules.jest.spec.ts]: Good spot — moved it into the shared `login-flow-inline-fixture` as `signInAndReadProfile`, taking the application and OIDC configuration as arguments so it isn't bound to one spec. Both now use the single copy (d5c81fbde3).

That does mean a small change to `idp-priority-order.jest.spec.ts` from #8456, since that's where the original lived — happy to pull it back to just this spec if you'd rather keep the PR narrower.

login-flow suite passes: 5 suites, 29 tests.

**Review by sclarkamlab — APPROVED** [review]
